### PR TITLE
Initialize the phpcr repository when testing the SonataDoctrinePHPCRAdminBundle

### PR DESCRIPTION
**Pull request by ElectricMaxxx**
So we are able to separate scripting in .travis folder to have
customized solutions like we would like to introduce in
[#535](https://github.com/sonata-project/SonataDoctrinePhpcrAdminBundle/pull/535).

### CODE REVIEW

File: project/.travis/before_script_test.sh.twig [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
 
+{% if 'phpcr' in services %}
+TESTING_SCRIPTS_DIR=vendor/symfony-cmf/testing/bin
+CONSOLE=${TESTING_SCRIPTS_DIR}/console
+
+export KERNEL_CLASS="Sonata\\DoctrinePHPCRAdminBundle\\Tests\\Fixtures\\App\\Kernel"
+echo '+++ create PHPCR +++'
+${CONSOLE} doctrine:phpcr:init:dbal --drop --force -vvv
+${CONSOLE} doctrine:phpcr:repository:init -vvv
+{% endif %}
```
Comment (OskarStark) [inline project/.travis/before_script_test.sh.twig]: please add a newline

File: project/.travis/before_script_test.sh.twig [hunk 2]
```diff
@@ -1,3 +1,12 @@
 #!/usr/bin/env bash
 set -ev
 
+{% if 'phpcr' in services %}
```
Comment (OskarStark) [inline project/.travis/before_script_test.sh.twig]: better check for project name here: doctrine-phpcr-admin-bundle

File: project/.travis/before_script_test.sh.twig [hunk 3]
```diff
@@ -1,3 +1,12 @@
...
+TESTING_SCRIPTS_DIR=vendor/symfony-cmf/testing/bin
```
Comment (OskarStark) [inline project/.travis/before_script_test.sh.twig]: no need for this var IMO

Comment (ElectricMaxxx) [reply project/.travis/before_script_test.sh.twig]: i will check that.

File: project/.travis/before_script_test.sh.twig (hunk 2 above)
Comment (ElectricMaxxx) [reply project/.travis/before_script_test.sh.twig]: according to the mongo service example you gave me, this would be consistent.

Comment (ElectricMaxxx) [reply project/.travis/before_script_test.sh.twig]: but having ci pipelines in mind, then services are additional software -  mostly containers, which is good for mongo, and here not :-1:

File: project/.travis/before_script_test.sh.twig [hunk 4]
```diff
@@ -1,3 +1,11 @@
 #!/usr/bin/env bash
 set -ev
 
+{% if repository_name is same as('SonataDoctrinePhpcrAdminBundle') %}
```
Comment (OskarStark) [inline project/.travis/before_script_test.sh.twig]: do we have access to `repository_name` here @Soullivaneuh ?

Comment (ElectricMaxxx) [reply project/.travis/before_script_test.sh.twig]: more then on `package_name` :-) i called the `dev-kit` in `dry-run` localy (i know that thing as i cloned it for cmf)

Comment (OskarStark) [reply project/.travis/before_script_test.sh.twig]: Fine for me

**Comment by dbu** [comment]
I edited the title.

File: project/.travis/before_script_test.sh.twig [hunk 5]
```diff
@@ -1,3 +1,11 @@
...
+CONSOLE=vendor/symfony-cmf/testing/bin/console
+
+export KERNEL_CLASS="Sonata\\DoctrinePHPCRAdminBundle\\Tests\\Fixtures\\App\\Kernel"
+echo '+++ create PHPCR +++'
+${CONSOLE} doctrine:phpcr:init:dbal --drop --force -vvv
+${CONSOLE} doctrine:phpcr:repository:init -vvv
+{% endif %}
```
Comment (greg0ire) [inline project/.travis/before_script_test.sh.twig]: Wouldn't it be nicer to just have something along the lines of
```bash
if [ -f before_script_test.local.sh ]
then
    ./before_script_test.local.sh
fi
```
(that's probably not the correct syntax / doesn't work from any working directory but you get the idea)

Comment (dbu) [reply project/.travis/before_script_test.sh.twig]: i think that would be better to separate responsibilities between the repos, yes.

do we have such "hooks" anywhere else? we should be consistent with them, ideally.

Comment (greg0ire) [reply project/.travis/before_script_test.sh.twig]: I don't think we do.

Comment (Soullivaneuh) [reply project/.travis/before_script_test.sh.twig]: We must take care to not be tempted to write quickly some script that could be common.

But yeah, I would also prefer a *.local.sh method for that case.

Comment (Soullivaneuh) [reply project/.travis/before_script_test.sh.twig]: BTW, the @greg0ire method looks good to me.

Maybe place the condition on both places:

- Before the main script execution
- At the end of the main script execution

WDYT?

Comment (dbu) [reply project/.travis/before_script_test.sh.twig]: yeah, if we have before_script_test.local.sh we should likely also have after_script_test.local.sh :-D

Comment (OskarStark) [reply project/.travis/before_script_test.sh.twig]: I created a PR https://github.com/sonata-project/dev-kit/pull/465, shall I change sth. ?

**Pull request closed without merging**